### PR TITLE
feat(private egress): update template and add log

### PR DESCRIPTION
**Pull request by karataliu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
To support private egress feature change in agentbaker. We plan to make it in three parts:
Part 1, adds data model, and test baseline
Part 2(this PR), update provision script to log value from data model, verify test basline change
Part 3, update real logic in provision script

**Which issue(s) this PR fixes**:

**Requirements**:
- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -945,6 +945,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"ShouldEnableCustomData": func() bool {
 			return !config.DisableCustomData
 		},
+		"PrivateEgressProxyAddress": func() string {
```
[the remark this answers is not in the document]
Comment (karataliu) [reply pkg/agent/baker.go]: fixed.

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -945,6 +945,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"ShouldEnableCustomData": func() bool {
 			return !config.DisableCustomData
 		},
+		"GetPrivateEgressProxyAddress": func() string {
+			return config.ContainerService.Properties.SecurityProfile.GetProxyAddress()
```
Comment (alexeldeib) [inline pkg/agent/baker.go:949]: nil checks? I think containerservice properties may be guaranteed to be populated but not security profile?

Comment (karataliu) [reply pkg/agent/baker.go:949]: Yes SecurityProfile could be nil, that nil check is covered in GetProxyAddress:
https://github.com/Azure/AgentBaker/blob/3a566a8e38b106fa107db271ab59257412feb62a/pkg/agent/datamodel/types.go#L2123

it's similar to GetUlimitConfig in https://github.com/Azure/AgentBaker/blob/3a566a8e38b106fa107db271ab59257412feb62a/pkg/agent/datamodel/types.go#L659

File: pkg/agent/baker_test.go [hunk 3]
```diff
@@ -1027,7 +1027,10 @@ oom_score = 0
 		Entry("AKSUbuntu2204 with SecurityProfile", "AKSUbuntu2204+SecurityProfile", "1.26.0",
 			func(config *datamodel.NodeBootstrappingConfiguration) {
 				config.ContainerService.Properties.SecurityProfile = &datamodel.SecurityProfile{
-					// TODO add PrivateEgress
+					PrivateEgress: &datamodel.PrivateEgress{
+						Enabled:      true,
+						ProxyAddress: "https://test-pe-proxy",
```
[the remark this answers is not in the document]
Comment (karataliu) [reply pkg/agent/baker_test.go:1032]: 1. yes, it's a real http proxy.
2. yes, aware there is a http proxy feature
3. This private egress proxy will be used in bootstrap script only:
e.g.
`curl --proxy <private-egress-proxy> <package to download>`
``

[the remark this answers is not in the document]
Comment (alexeldeib) [reply pkg/agent/baker_test.go:1032]: consider port + scheme (http/https) handling if you have a single endpoint.

**Review by alexeldeib — APPROVED** [review]

**Review by AbelHu — APPROVED** [review]
LGTM

File: pkg/agent/baker_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (karataliu) [reply pkg/agent/baker_test.go:1032]: sure will. thanks.

**Pull request merged**